### ISSUE TITLE
Is approach not working with ios anymore ?

### ISSUE THREAD

**Issue by d1sna** [choutkamartin/next-auth-capacitor#7]
There is no possible to set "http://" or "https://" instead of "capacitor://" in requests to next-auth, therefore origin and host are different in requests to next-auth and providers urls are not exist in response by server. Do you know about this? is it possible to manage it anyway?

**Comment by felixpahlke** [comment]
> There is no possible to set "http://" or "https://" instead of "capacitor://" in requests to next-auth, therefore origin and host are different in requests to next-auth and providers urls are not exist in response by server. Do you know about this? is it possible to manage it anyway?

Have you found a solution to this? Am currently trying to make nextauth work with capacitor and I am having the same problems.

**Comment by d1sna** [comment]
> > There is no possible to set "http://" or "https://" instead of "capacitor://" in requests to next-auth, therefore origin and host are different in requests to next-auth and providers urls are not exist in response by server. Do you know about this? is it possible to manage it anyway?
> 
> Have you found a solution to this? Am currently trying to make nextauth work with capacitor and I am having the same problems.

Yes, I've done it, but it was really difficult and not best solution, better to create custom auth for mobile with jwt. 
There are some points that i remeber now: 

1. Fetch is not working fine with mobile, especially for auth with credentials (cookie), instead fetch lib u need to use htttp-capacitor official plugin 
2. UseSession from next-auth doesn't have possibility to send request from capacitor://yourdomain to https://yourdomain and you need to create custom UseSession 
3. I created custom provider in next-auth for auth mobile users, used native auth (google, facebook, etc) + custom auth route to authorize, then stored session and cookie in local storage and set it at application start

**Comment by felixpahlke** [comment]
Thanks for your reply!! By UseSession() you mean the actual useSession() hook that retrieves the session data in a component? By Custom Provider do you mean A Custom SessionProvider Context or a custom provider as in "EmailProvider" "GoogleProvider" etc?

**Comment by d1sna** [comment]
> Thanks for your reply!! By UseSession() you mean the actual useSession() hook that retrieves the session data in a component? By Custom Provider do you mean A Custom SessionProvider Context or a custom provider as in "EmailProvider" "GoogleProvider" etc?

yep, you could not use useSession hook "from the box". 

 By Custom Provider do you mean A Custom SessionProvider Context or a custom provider as in "EmailProvider" "GoogleProvider" etc?
 
 both, i used custom credentialsProvider (in next-auth file) + native google auth (capacitor plugin)/fb/etc to authroize users, not good solution, but i didnt find another way
